autoload/lsp.vim

```diff
@@ -214,7 +217,30 @@ function! s:call_did_save(buf, server_name, result, cb) abort
 endfunction
 
 function! s:on_text_document_did_close() abort
-    call lsp#log('s:on_text_document_did_close()', bufnr('%'))
+    let l:buf = bufnr('%')
+    call lsp#log('s:on_text_document_did_close()', l:buf)
+endfunction
+
+function! lsp#get_last_file_content(server_name, buf) abort
+    if has_key(s:file_content, a:buf) && has_key(s:file_content[a:buf], a:server_name)
+        return s:file_content[a:buf][a:server_name]
+    endif
+    return []
+endfunction
+
+function! lsp#update_file_content(server_name, buf, new) abort
+    if !has_key(s:file_content, a:buf)
+        let s:file_content[a:buf] = {}
+    endif
+    let s:file_content[a:buf][a:server_name] = a:new
+endfunction
+
+function! s:on_buf_wipeout(buf) abort
+    if has_key(s:file_content, a:buf)
```

Comment:
why not just remove the entire `s:file_content` for that buf though current approach is also ok.